src/pages/index.js

```diff
@@ -117,32 +165,92 @@ function handleCloseModal(modal) {
   disableEsc();
 }
 
+function returnToSave(button) {
+  setTimeout(() => {
+    button.textContent = "Save";
+  }, 2000);
+}
+
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  handleCloseModal(profileFormElement);
+  profileSubmitButton.textContent = "Saving...";
+  const patch = { name: nameInput.value, about: jobInput.value };
+  api
+    .editUserInfo(patch)
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileJobElement.textContent = data.about;
+      handleCloseModal(profileFormElement);
+      returnToSave(profileSubmitButton);
+    })
+    .catch((err) => console.error(err));
 }
 
-function renderCards() {
-  initialCards.forEach((card) => {
-    cardsListElement.append(getCardElementData(card));
-  });
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  newPostSubmitButton.textContent = "Saving...";
+  api
+    .postNewCard(captionInput.value, linkInput.value)
+    .then((res) => {
+      const cardElement = getCardElementData(res);
+      cardsListElement.prepend(cardElement);
+      evt.target.reset();
+      disableButton(newPostSubmitButton, settings);
+      handleCloseModal(newPostFormElement);
+      returnToSave(newPostSubmitButton);
+    })
+    .catch((err) => {
+      console.log("Error creating card:", err);
+    });
 }
 
-function renderNewCard(card) {
-  cardsListElement.prepend(getCardElementData(card));
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  avatarSubmitButton.textContent = "Saving...";
+  api
+    .editUserAvatar(avatarInput.value)
+    .then((data) => {
+      profileImg.src = data.avatar;
+      handleCloseModal(avatarModal);
+      returnToSave(avatarSubmitButton);
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails. 



**Please, correct it everywhere in the code**